sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -243,7 +243,7 @@ partitionSpec
     ;
 
 partitionVal
-    : identifier (EQ constant)?
+    : identifier (comparisonOperator constant)?
```

Comment:
@dongjoon-hyun that is an interesting point. We might need to take a step back here. The only thing I want to prevent from happening is that we are going to recreate an expression hierarchy, and not reuse what is already there. That being said, things must remain concise. So if it takes a large amount of time or code, then we should reconsider.

In this case I would like to try the expression approach first if you don't mind.